lib/variable/include/scipp/variable/variable.tcc

```diff
@@ -38,12 +38,21 @@ template <class T> auto &cast(Variable &var) {
 }
 
 template <int I, class T> decltype(auto) get(T &&t) {
-  if constexpr (core::has_eval_v<std::decay_t<T>>)
-    return t.operator()(I);
-  else if constexpr (std::is_same_v<std::decay_t<T>, Eigen::Affine3d>)
+  if constexpr (std::is_same_v<std::decay_t<T>, Eigen::Affine3d>) {
     return t.matrix().operator()(I);
-  else
+  } else if constexpr (std::is_same_v<std::decay_t<T>,
+                                      scipp::core::ScalingTransform> ||
+                       std::is_same_v<std::decay_t<T>,
+                                      scipp::core::RotationTransform>) {
+    return t.matrix().operator()(I);
+  } else if constexpr (std::is_same_v<std::decay_t<T>,
+                                      scipp::core::TranslationTransform>) {
+    return t.vector().operator()(I);
+  } else if constexpr (core::has_eval_v<std::decay_t<T>>) {
+    return t.operator()(I);
+  } else {
```

Comment:
At least some of the `matrix()` methods return by value, so I feel bad things will/may happen here. I think this can probably be avoided entirely, just expose access to the actual components, not to the equivalent matrices or vectors.